web/ui/dashboard/src/app/components/modal/modal.component.html

```diff
@@ -1,37 +1,10 @@
 <div convoyOverlay [overlayHasBackdrop]="true" (click)="closeModal.emit()"></div>
 <div convoy-modal-dialog [id]="id" [position]="position" [size]="size">
-	<div *ngIf="title" class="px-20px pt-20px pb-16px border-b border-b-grey-10 bg-white-100 rounded-tr-16px rounded-tl-16px">
-		<div class="flex items-center justify-between" *ngIf="position !== 'full'">
-			<h2 class="font-semibold capitalize">{{ title }}</h2>
+	<ng-content select="[modalHeader]"></ng-content>
 
-			<button convoy-button texture="light" size="sm" class="px-8px !py-10px" (click)="closeModal.emit()">
-				<img src="/assets/img/modal-close-icon.svg" alt="close icon" />
-			</button>
-		</div>
-
-		<div class="flex items-center justify-between m-auto max-w-[834px]" *ngIf="position === 'full'">
-			<div class="flex items-center">
-				<button convoy-button size="sm" texture="light" class="px-10px !py-10px" (click)="closeModal.emit()">
-					<img src="/assets/img/modal-close-icon.svg" class="w-12px" alt="close icon" />
-				</button>
-
-				<h2 class="font-semibold capitalize ml-2">{{ title }}</h2>
-			</div>
-
-			<a class="flex items-center" target="_blank" href="https://getconvoy.io/docs" rel="noreferrer">
-				<img src="/assets/img/doc-icon-primary.svg" alt="doc icon" />
-				<span class="font-medium text-14 text-primary-100 ml-2">Go to docs</span>
-			</a>
-		</div>
-	</div>
-
-	<div class="m-auto empty:hidden" [class]="position === 'full' ? 'max-w-[834px]' : 'w-full p-20px'">
-		<ng-content select="[modalBody]"></ng-content>
-	</div>
-
-	<div class="fixed w-full bottom-0 px-20px pt-20px pb-16px border-t border-t-grey-10 empty:hidden">
-		<ng-content select="[modalFooter]"></ng-content>
-	</div>
+    <div convoy-modal-body [position]="position">
+        <ng-content select="[modalBody]"></ng-content>
```

Comment:
I'm not sure we need to do this; think of it like this:
- Modal component is a container
- It knows all the possible shapes we might want our modal to have
- The only thing we want to pass into the ng-content when using the component is the modal title and body together, not separately. Why: there is no reason to have them separate.
```html
<div convoy-modal>
    <div convoy-modal-title></div>
    <div convoy-modal-body></div> // not sure if we'll need a directive like this though
</div>
```
- To close the modal, we should import an instance of the modal component and call a close modal method